bundles/org.openhab.binding.smaenergymeter/src/main/java/org/openhab/binding/smaenergymeter/internal/discovery/SMAEnergyMeterDiscoveryService.java

```diff
@@ -54,35 +65,48 @@ public Set<ThingTypeUID> getSupportedThingTypes() {
 
     @Override
     protected void startBackgroundDiscovery() {
+        PacketListener packetListener = this.packetListener;
+        if (packetListener != null) {
+            return;
+        }
+
         logger.debug("Start SMAEnergyMeter background discovery");
-        scheduler.schedule(this::discover, 0, TimeUnit.SECONDS);
+        try {
+            packetListener = listenerRegistry.getListener(PacketListener.DEFAULT_MCAST_GRP,
+                    PacketListener.DEFAULT_MCAST_PORT);
+            packetListener.open(30);
+        } catch (IOException e) {
+            logger.warn("Could not start background discovery", e);
+            return;
+        }
+
+        packetListener.addPayloadHandler(this);
+        this.packetListener = packetListener;
     }
 
     @Override
-    public void startScan() {
-        logger.debug("Start SMAEnergyMeter scan");
-        discover();
+    protected void stopBackgroundDiscovery() {
+        PacketListener packetListener = this.packetListener;
+        if (packetListener != null) {
+            packetListener.removePayloadHandler(this);
```

Comment:
I guess it would be clean to also set this.packetListener  to null ?

Reply:
done